### PR TITLE
1. Comparable with vim7.4. 2. support nvim built-in lua funtion

### PR DESCRIPTION
**Pull request by leoatchina**
1.   Comparable with vim7.4 by adding missing functions when `has('lambda')`, `execute()` not exist
2.  Support nvim builtin-in lua function: The function name is layed in the runtime lua script, and raw.rhs contain the name and the line of the funtion, so it is not hard to parser the function.

### CODE REVIEW

File: autoload/which_key/mappings.vim [hunk 1]
```diff
@@ -43,7 +54,9 @@ function! which_key#mappings#parse(key, dict, visual) " {{{
   if key ==# '<Tab>'
     call extend(lines, s:get_raw_map_info('<C-I>'))
   endif
-
+  if !has('nvim') && key[0:2] == '<M-' && !has('patch-8.2.0815')
```
Comment (liuchengxu) [inline autoload/which_key/mappings.vim:57]: Can you please add a comment here?

File: autoload/which_key/mappings.vim [hunk 2]
```diff
@@ -43,21 +54,41 @@ function! which_key#mappings#parse(key, dict, visual) " {{{
...
+        let fl = expand(sp[0])
+        " NOTE: ln is the line where lua function exists
+        let ln = str2nr(sp[-1]) - 1
+        let rhs = trim(readfile(fl)[ln])
+        let rhs = split(rhs, 'M.')[1]
+        " create api from file name
+        let api = split(substitute(fl, "\\", "/", 'g'), 'runtime/lua/')[1]
+        let api = substitute(api, 'lua$', '', 'g')
+        let api = substitute(api, '/', '.', 'g')
+        let mapd.rhs = "<Cmd>lua " . api . rhs . '<Cr>'
```
Comment (liuchengxu) [inline autoload/which_key/mappings.vim:85]: @burnettk  Do you mind having a try with this PR regarding the lua functions?

File: autoload/which_key/mappings.vim [hunk 3]
```diff
@@ -43,21 +54,41 @@ function! which_key#mappings#parse(key, dict, visual) " {{{
   if key ==# '<Tab>'
     call extend(lines, s:get_raw_map_info('<C-I>'))
   endif
-
+  if !has('nvim') && key[0:2] == '<M-' && !has('patch-8.2.0815')
+    let key = eval('"\' . key . '"')
+  endif
   for line in lines
     " filter out lines like: n  <Space>ca   *@<Lua 129: /opt/homebrew/Cellar/neovim/0.9.0/share/nvim/runtime/lua/vim/lsp/buf.lua:758>
     " we're not going to get anything useful to display from the rhs of these anyway
```
Comment (liuchengxu) [inline autoload/which_key/mappings.vim:62]: Outdated, and worth a new comment as the parsing below is not straightforward.

**Review by liuchengxu — COMMENTED** [review]
If you can separate out the first commit into another PR, we can merge it soon.

My concern about the lua function parsing is whether the format is stable enough, but I'm happy to merge it once @burnettk can confirm.

File: autoload/which_key/mappings.vim (hunk 2 above)
Comment (burnettk) [reply autoload/which_key/mappings.vim:85]: @liuchengxu it seems to work and not cause a regression with the issue i was seeing before. thanks for checking.

File: .gitignore
Comment (liuchengxu) [inline .gitignore:1]: This change is unrelated.

**Review by liuchengxu — COMMENTED** [review]
Revert the change on .gitignore and then we're good to go.

**Comment by leoatchina** [comment]
I think I should close this pr and refork and repr
